From b0f8ddd63df1dd8c563bd0b96c48a2e7c7780747 Mon Sep 17 00:00:00 2001
From: Jean-Christophe Trotin <jean-christophe.trotin@st.com>
Date: Thu, 20 Oct 2016 18:17:28 +0200
Subject: [PATCH 17/23] v4l2object: fill colorimetry in
 gst_v4l2_object_acquire_format

Instead of relying on the default colorimetry chosen by
gst_video_info_set_format(), set info.colorimetry from the
values returned by G_FMT. This allows decoders to propagate
their input colorimetry downstream.

https://bugzilla.gnome.org/show_bug.cgi?id=766383

Change-Id: I22b0f92926ae9aac0ed3710584e2c71f3952e441
Reviewed-on: https://gerrit.st.com/59413
Tested-by: Jean Christophe TROTIN <jean-christophe.trotin@st.com>
Reviewed-by: Jean Christophe TROTIN <jean-christophe.trotin@st.com>
---
 sys/v4l2/gstv4l2object.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/sys/v4l2/gstv4l2object.c b/sys/v4l2/gstv4l2object.c
index 1347df4..deab718 100644
--- a/sys/v4l2/gstv4l2object.c
+++ b/sys/v4l2/gstv4l2object.c
@@ -3697,6 +3697,8 @@ gst_v4l2_object_acquire_format (GstV4l2Object * v4l2object, GstVideoInfo * info)
       goto unsupported_field;
   }
 
+  gst_v4l2_object_get_colorspace (&fmt, &info->colorimetry);
+
   gst_v4l2_object_save_format (v4l2object, fmtdesc, &fmt, info, &align);
 
   /* Shall we setup the pool ? */
-- 
2.7.4

